src/libpriv/rpmostree-core.c

```diff
@@ -2560,20 +2736,8 @@ delete_package_from_root (RpmOstreeContext *self,
       if (!g_str_has_prefix (fn, "usr/"))
         continue;
 
-      /* avoiding the stat syscall is worth a bit of userspace computation */
-      if (rpmostree_str_has_prefix_in_ptrarray (fn, deleted_dirs))
-        continue;
```

Comment:
I dropped this optimization here since we now do things like librpm does: we traverse the file list backwards, which means directories are emptied out before we delete them.